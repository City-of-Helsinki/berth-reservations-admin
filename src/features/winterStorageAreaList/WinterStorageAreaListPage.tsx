import React from 'react';
import { useTranslation } from 'react-i18next';

import Table, { Column, COLUMN_WIDTH } from '../../common/table/Table';
import styles from './winterStorageAreaListPage.module.scss';
import { WinterStorageAreaData } from './types';
import PageTitle from '../../common/pageTitle/PageTitle';

type ColumnType = Column<WinterStorageAreaData> & { accessor: keyof WinterStorageAreaData };

export type WinterStorageAreaListPageProps = {
  data: Array<WinterStorageAreaData>;
  loading?: boolean;
};

const WinterStorageAreaListPage = ({ data, loading }: WinterStorageAreaListPageProps) => {
  const { t } = useTranslation();

  const columns: ColumnType[] = [
    {
      Header: t('winterStorageAreaList.tableHeaders.name') || '',
      accessor: 'name',
      width: COLUMN_WIDTH.S,
    },
  ];

  return (
    <div className={styles.winterStorageAreaListPage}>
      <PageTitle title={t('winterStorageAreaList.title')} />
      <Table data={data} loading={loading} columns={columns} />
    </div>
  );
};

export default WinterStorageAreaListPage;
