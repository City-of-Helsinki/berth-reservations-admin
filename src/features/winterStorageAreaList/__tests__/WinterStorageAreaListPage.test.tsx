import React from 'react';
import { shallow } from 'enzyme';

import WinterStorageAreaListPage from '../WinterStorageAreaListPage';

const mockProps = {
  data: [
    {
      id: '0',
      name: 'Test WS Area 0',
    },
    {
      id: '1',
      name: 'Test WS Area 1',
    },
  ],
};

describe('WinterStorageAreaListPage', () => {
  const getWrapper = (props = {}) => shallow(<WinterStorageAreaListPage {...mockProps} {...props} />);

  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.render()).toMatchSnapshot();
  });
});
