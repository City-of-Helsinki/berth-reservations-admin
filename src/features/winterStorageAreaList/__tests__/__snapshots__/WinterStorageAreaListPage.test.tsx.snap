// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinterStorageAreaListPage renders normally 1`] = `
<div
  class="winterStorageAreaListPage"
>
  <h2
    class="text standard h2 xl normalWeight pageTitle"
  >
    TALVISÄILYTYSALUEET
  </h2>
  <div>
    <div
      class="tableWrapper"
    >
      <div
        class="table primary"
        role="table"
        style="min-width:112.5px"
      >
        <div
          class="stickyHeaders"
        >
          <div
            class="header mainHeader"
            role="row"
            style="display:flex;flex:1 0 auto;min-width:0px"
          >
            <div
              class="headerCell"
              colspan="1"
              role="columnheader"
              style="box-sizing:border-box;flex:112.5 0 auto;min-width:0px;width:112.5px"
            >
              Talvisäilytysalueet
            </div>
          </div>
          <div
            class="header"
            role="row"
            style="display:flex;flex:1 0 auto;min-width:0px"
          >
            <div
              class="headerCell"
              colspan="1"
              role="columnheader"
              style="box-sizing:border-box;flex:112.5 0 auto;min-width:0px;width:112.5px;cursor:pointer"
              title="Toggle SortBy"
            >
              Alue
              <div
                class="sortArrowWrapper"
              >
                <svg
                  class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_m__3edUY icon_hds-icon--size-m__1mcHv sortArrow sortArrowDown"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fill-rule="evenodd"
                  >
                    <path
                      d="M0 0h24v24H0z"
                    />
                    <path
                      d="M10 5.636l1.414 1.414L7.465 11H20v2H7.464l3.95 3.95L10 18.364 3.636 12z"
                      fill="currentColor"
                    />
                  </g>
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          role="rowgroup"
        >
          <div
            class="rowWrapper"
          >
            <div
              role="row"
              style="display:flex;flex:1 0 auto;min-width:0px"
            >
              <div
                class="tableCell"
                role="cell"
                style="box-sizing:border-box;flex:112.5 0 auto;min-width:0px;width:112.5px"
              >
                Test WS Area 0
              </div>
            </div>
          </div>
          <div
            class="rowWrapper"
          >
            <div
              role="row"
              style="display:flex;flex:1 0 auto;min-width:0px"
            >
              <div
                class="tableCell"
                role="cell"
                style="box-sizing:border-box;flex:112.5 0 auto;min-width:0px;width:112.5px"
              >
                Test WS Area 1
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
